### PR TITLE
UWP (Win10) support

### PR DESCRIPTION
**Pull request by niklaswolf**
Hi @endel, as I promised in issue #8, I adapted the Client to also support UWP (Windows 10). It was a bit tricky, but I managed to get it working.
I have to mention, that this is only tested in Unity directly and on Microsoft Hololens, which is running UWP apps (Win10). But because UWP is the same set of APIs for other devices, it should work for them too.

Some important changes:
- I had to get rid of WebSocketSharp, so I wrapped it in preprocessor-directives to not include it when the app is build with an UWP-app as target, hope this is okay.
- Added another section to WebSocket.cs which uses the UWP-APIs for connection etc.
- had to add another MessagePack.dll for UWP, named it MessagePackUWP. I then edited the import-Settings in Unity, so that the right dll gets included for the different target-platforms.
- added some public fields to ColyseusClient.cs, so that you can set servername, port and roomname directly from Unity

I don't know, if you want to merge this into the General Unity-client, or if it should stand alone from now on...
I also will have to write a little guide for building this for UWP, as I got some Errors which can be fixed by following a specific build process.

So what do you think? :)

### CODE REVIEW

File: Assets/Colyseus/Room.cs [hunk 1]
```diff
@@ -76,9 +79,13 @@ public int id
...
 			set {
 				this._id = value;
-				this.OnJoin.Emit (this, EventArgs.Empty);
-			}
-		}
+#if !WINDOWS_UWP
+                this.OnJoin.Emit (this, EventArgs.Empty);
+#else
+                this.OnJoin.Invoke(this, EventArgs.Empty);
+#endif
```
Comment (endel) [inline Assets/Colyseus/Room.cs]: Looks like `Emit` is an alias to `Invoke`. I didn't know that when I first write this. We can use always `Invoke`. Can we get rid of the conditional compilation?

File: Assets/Colyseus/Room.cs [hunk 2]
```diff
@@ -87,24 +94,34 @@ public void SetState( MessagePackObject state, int remoteCurrentTime, int remote
 
 			// TODO:
 			// Create a "clock" for remoteCurrentTime / remoteElapsedTime to match the JavaScript API.
-
+
```
Comment (endel) [inline Assets/Colyseus/Room.cs]: No white spaces, please :)

File: Assets/Colyseus/Room.cs [hunk 3]
```diff
@@ -87,24 +94,34 @@ public void SetState( MessagePackObject state, int remoteCurrentTime, int remote
...
 			var serializer = MessagePackSerializer.Get <MessagePackObject>();
-			this._previousState = serializer.PackSingleObject (state);
 
-			this.OnUpdate.Emit (this, new RoomUpdateEventArgs (this, state, null));
-		}
+#if !WINDOWS_UWP
+            this.OnUpdate.Emit(this, new RoomUpdateEventArgs(this, state, null));
+#else
+            this.OnUpdate.Invoke(this, new RoomUpdateEventArgs(this, state, null));
+#endif
```
Comment (endel) [inline Assets/Colyseus/Room.cs]: Use same indentation

File: Assets/Colyseus/Client.cs [hunk 4]
```diff
@@ -86,22 +88,33 @@ public void Recv()
...
 			UnpackingResult<MessagePackObject> raw = Unpacking.UnpackObject (recv);
 
 			var message = raw.Value.AsList ();
 			var code = message [0].AsInt32 ();
 
-			Console.WriteLine(code);
+#if !WINDOWS_UWP
+            Console.WriteLine(code);
+#else
+            Debug.Log("Message Code: "+code);
```
Comment (endel) [inline Assets/Colyseus/Client.cs]: We can remove this log completely. It shouldn't be there in the first place :)

File: Assets/Colyseus/Client.cs [hunk 5]
```diff
@@ -86,22 +88,33 @@ public void Recv()
...
+#endif
 
-			// Parse roomId or roomName
-			Room room = null;
+            // Parse roomId or roomName
+            Room room = null;
```
Comment (endel) [inline Assets/Colyseus/Client.cs]: indentation

File: Assets/ColyseusClient.cs [hunk 6]
```diff
@@ -7,15 +7,19 @@
 public class ColyseusClient : MonoBehaviour {
 
 	Client colyseus;
-	Room chatRoom;
+    Room chatRoom;
+    public string serverName = "localhost";
+    public string port = "3553";
+    public string roomName = "chat";
```
Comment (endel) [inline Assets/ColyseusClient.cs]: Please also review indentation of this whole file.

**Comment by endel** [comment]
Hey @niklaswolf, did you had a chance to have look at this?

**Comment by niklaswolf** [comment]
Hi @endel, yep I have read your comments, will have some time implementing it this weekend ;)
Sorry for not answering, but I was a bit busy ;)

**Comment by endel** [comment]
No worries! Great, thanks!

**Comment by niklaswolf** [comment]
Hi @endel, I have changed the things you requested. Maybe the indentation is still bad, but I don't understand why Visual Studio is showing different (correct) indentation than other editors. I tried my best ;)

I also added a little section to the Readme with build instructions for UWP. Additionally I fixed the Readme markup for the c#-code, so now it should show up properly again ;)

**Comment by endel** [comment]
Awesome, thanks for reviewing. I've invited you to the organization, feel free to tidy things up here and send more pull-requests if you want to :)

**Pull request merged**